docs/Course Wiki/ECE Course Offerings/ECE374B.md

```diff
@@ -0,0 +1,62 @@
+# ECE374B
+## Introduction to Algorithms and Models of Computation
+
+ECE374B is one of two **independent** sections of CS/ECE374, Introduction to Algorithms and Models of Computation. It is unoficially the 'ECE' version of 374. It is considered by some to be easier than the A version.
+
+## Content Covered
+
+#### Theory of Computation 
+- Regular and context-free languages.
+- Fooling sets.
+- Determinisitc and non-deterministic finite state automata.
+- Push down automata
+- Turing Machines
+- Computational Complexity
+#### Algorithms
+- Divide & Conquer
+- Graph Algorithms
+- Shortest Paths, MSTs
+- Dynamic Programming
+- Directed Acylic Graphs
+- Algorithmic Complexity Analysis
+#### Secret Third Thing
+- Reductions
+- Undecidability
+- NP, NP-Hardness, P vs NP
+
+## Prerequisites
+- [CS 173](../CS%20Course%20Offerings/CS173.md)
+- [CS 225](../CS%20Course%20Offerings/CS225.md)
+
+There are two official prerequisites to ECE 374, CS 173 - Discrete Structures, and CS 225 - Data Structures and Algorithms. 374B expands on the theoretical computer science from CS173 and covers algorithms more in depth than CS225, without any of the programming. For part one of the course, you should remember induction, set notation, and FSMs from CS 173. Induction and recursion are fundamental to the course, so it is important to know them well. In the second portion of the course, there is a significant overlap with the graph algorithms taught in CS225.
+
+## When to Take it
+ECE374 is a required course for all Computer Engineering and Computer Science majors. ECE 374 is not used as a prerequisite for any course in the ECE department, so for students not interested in higher level CS courses, there is little reason to rush it. However, due to the knowledge needed from CS 225 and CS 173, it might help to take it within a few semesters after finishing CS 225. On the other hand, taking this course early would help to prepare for Dynamic Programming interview questions. It is very common to encouter DP questions when applying for software engineering internships, and the HWs in ECE374 help prepare for them well. This course is fairly challenging so take caution if you plan to take this course with too many other technical courses.
```

Comment:
Delete the space between "ECE" and "374". Should be "ECE374". Apply this everywhere.
Second sentence: Replace first ECE374 with "It".
"Taking this course early would help prepare for software interview questions, particularly those related to Dynamic Programming".
A bit more detail is needed on how to rush this course (just mention that one should take 225 asap).